lib/prism/parse_result.rb

```diff
@@ -143,6 +143,14 @@ def code_units_column(byte_offset, encoding)
       code_units_offset(byte_offset, encoding) - code_units_offset(line_start(byte_offset), encoding)
     end
 
+    # Returns the byte offset for a given line number and column number
+    def line_and_character_column_to_byte_offset(line, column)
+      line_start = offsets[line - 1]
+      line_end = offsets[line]
+      byte_column = (@source.byteslice(line_start, line_end) or raise)[0...column]&.bytesize #: Integer
```

Comment:
The `raise` here might be a bit hacky, since e.g. if the file shrinked we might end up with `nil` for `byteslice`.
Should we raise a better error, or return `nil` in such a case?
I think an better error (e.g. ArgumentError with some telling message) seems better and less confusing.

`[0...column]` is fine because that always returns a String since it starts at 0: `""[0...42] => ""`.